### PR TITLE
backlog/6_outputting_to_ckan

### PR DESCRIPTION
**Pull request by frbattid**
- Implements the issue https://github.com/telefonicaid/fiware-tidoop/issues/6
- 100% unit tests passed

```
Tests run: 6, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.029 sec
Results :
Tests run: 34, Failures: 0, Errors: 0, Skipped: 0
```
- (unofficial) e2e tests passed: http://130.206.127.30:4000/dataset/testnoadditionalinfo2/resource/daa26b2f-3eaa-49ac-802e-7446883e4bf0
- Assignee @gtorodelvalle

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -263,13 +307,13 @@ Finally, the application can be run as:
...
         <comma-separated list of ckan inputs> \
-        <hdfs output folder> \
+        <ckan output package> \
         <splits length>
         
 An example of run:
 
 ```
-$ hadoop jar /home/user1/fiware-tidoop/target/tidoop-0.0.0_SNAPSHOT-jar-with-dependencies.jar com.telefonica.iot.tidoop.utils.CKANMapReduceExample -libjars /home/user1/fiware-tidoop/target/tidoop-0.0.0_SNAPSHOT-jar-with-dependencies.jar -D mapreduce.job.queuename=user1q data.lab.fiware.org 443 true 2d5bf021-ff9f-48e3-bb97-395b77581665 https://data.lab.fiware.org/dataset/logrono_cygnus/resource/ca73a799-9c71-4618-806e-7bd0ca1911f4 /user/user1/ckan_output 1000
+$ hadoop jar /home/user1/fiware-tidoop/target/tidoop-0.0.0_SNAPSHOT-jar-with-dependencies.jar com.telefonica.iot.tidoop.utils.CKANMapReduceExample -libjars /home/user1/fiware-tidoop/target/tidoop-0.0.0_SNAPSHOT-jar-with-dependencies.jar data.lab.fiware.org 443 true xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx https://data.lab.fiware.org/dataset/logrono_cygnus/resource/ca73a799-9c71-4618-806e-7bd0ca1911f4 https://data.lab.fiware.org/dataset/logrono_cygnus 1000
```
Comment (gtorodelvalle) [inline README.md:316]: My eyes just went straight to the xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx :D and it was quite "complex" to check that it referred to the <ckan API key> option... Maybe a comment in the "An example of run:" line changing it to something such as "An example of run (using a fake CKAN API 'xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx'):" may help... ;) No need to say that it is not really important... :)

File: src/main/java/com/telefonica/iot/tidoop/backends/ckan/CKANBackend.java [hunk 2]
```diff
@@ -198,6 +210,110 @@ public JSONArray getRecords(String resId, long start, long end) {
...
+                throw new Exception("Don't know how to treat the response code. Possibly the datastore "
+                        + "already exists (respCode=" + res.getStatusCode() + ", resourceId=" + resId + ")");
+            } // if else if else
+        } catch (Exception e) {
+            throw new Exception(e.getMessage());
+        } // try catch
+    } // createKeyValueDatastore
+    
+    /**
+     * Inserts a (key, value) pair into a givern resource.
```
Comment (gtorodelvalle) [inline src/main/java/com/telefonica/iot/tidoop/backends/ckan/CKANBackend.java]: Typo s/givern/given

**Comment by gtorodelvalle** [comment]
Not much to say apart from the previous minor comments :) Nice patch! ;)

File: README.md (hunk 1 above)
Comment (frbattid) [inline README.md:316]: I agree, it was a really rare to see such a buch of "xxxxx" :) Fixed in https://github.com/telefonicaid/fiware-tidoop/commit/43a8b52155e83e6cc2096b18679049cf607f2a62

File: src/main/java/com/telefonica/iot/tidoop/backends/ckan/CKANBackend.java (hunk 2 above)
Comment (frbattid) [inline src/main/java/com/telefonica/iot/tidoop/backends/ckan/CKANBackend.java]: Fixed in https://github.com/telefonicaid/fiware-tidoop/commit/43a8b52155e83e6cc2096b18679049cf607f2a62

**Comment by gtorodelvalle** [comment]
Thanks @frbattid ! ;) LGTM aaand merging! :)

**Pull request merged**